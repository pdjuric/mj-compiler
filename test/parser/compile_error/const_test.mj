program main_test

class T{}

const char a = 5, b = true;
const bool c = 'a', d = 3;
const int e = 's', f = false;
const T t = 5;

{
    void main(){}
}