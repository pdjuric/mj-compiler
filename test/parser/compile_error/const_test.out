ERROR 15:50:08,081 - line   5 - Constant a must be initialized with constant of type int
ERROR 15:50:08,081 - line   5 - Constant b must be initialized with constant of type bool
ERROR 15:50:08,081 - line   6 - Constant c must be initialized with constant of type char
ERROR 15:50:08,082 - line   6 - Constant d must be initialized with constant of type int
ERROR 15:50:08,082 - line   7 - Constant e must be initialized with constant of type char
ERROR 15:50:08,082 - line   7 - Constant f must be initialized with constant of type bool
ERROR 15:50:08,082 - line   8 - Constant type is not int, char or bool.